pkg/utils/errgroup.go

```diff
@@ -5,54 +5,45 @@ import (
 	"sync/atomic"
 )
 
-type FirstErrGroup struct {
-	errChan  chan error
+type ErrorGroup struct {
```

Comment:
Why changing the `FirstErrorGroup`? The original logic is also return the first error when initializing multiple worker nodes.
refer to https://github.com/cnrancher/autok3s/pull/491/commits/33b347c1618120b4c984dfa8c0fca533b6b63bfa#diff-f4933755bd63c9e160a2068e8ccb4048e8df1b43f7f4544dcbc7ae2650a07478L140

Reply:
Confirmed with @Jason-ZW that the original logic when initializing multiple worker nodes should catch all errors after all goroutines are complete. 
The original design is to catch all initializing errors to fix the problem once. If we only catch and return the first error, users can't fix all problems and will join nodes multiple times to fix them one by one. 
If we want to return the first error, I think it's better to skip all goroutines instead of for waiting all of them to complete and then getting the first error.

So I changed the FirstErrGroup to ErrorGroup to catch all errors. Next time I will enhance the join nodes logic to only rollback nodes with error instead of rollback all.